x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/CrossClusterAccessAuthenticationService.java

```diff
@@ -170,25 +148,38 @@ private void validate(final CrossClusterAccessSubjectInfo crossClusterAccessSubj
             );
         }
 
-        for (CrossClusterAccessSubjectInfo.RoleDescriptorsBytes roleDescriptorsBytes : crossClusterAccessSubjectInfo
-            .getRoleDescriptorsBytesList()) {
-            final Set<RoleDescriptor> roleDescriptors = roleDescriptorsBytes.toRoleDescriptors();
-            for (RoleDescriptor roleDescriptor : roleDescriptors) {
-                final boolean privilegesOtherThanIndex = roleDescriptor.hasClusterPrivileges()
-                    || roleDescriptor.hasConfigurableClusterPrivileges()
-                    || roleDescriptor.hasApplicationPrivileges()
-                    || roleDescriptor.hasRunAs()
-                    || roleDescriptor.hasRemoteIndicesPrivileges();
-                if (privilegesOtherThanIndex) {
-                    throw new IllegalArgumentException(
-                        "role descriptor for cross cluster access can only contain index privileges "
-                            + "but other privileges found for subject ["
-                            + effectiveSubject.getUser().principal()
-                            + "]"
-                    );
+        final User user = effectiveSubject.getUser();
+        if (CrossClusterAccessUser.is(user)) {
+            // Assert here as an internal consistency check; the role descriptors will be subsequently ignored so no need to fail validation
+            // with an exception
+            assert crossClusterAccessSubjectInfo.getRoleDescriptorsBytesList().isEmpty()
+                : "received role descriptors bytes list should be empty for internal cross cluster access user";
```

Comment:
We probably should log this (maybe even at warn level) so that the behaviour is not entirely silent.